src/core/lib/transport/metadata_batch.cc

```diff
@@ -166,14 +163,12 @@ grpc_error* grpc_metadata_batch_link_head(grpc_metadata_batch* batch,
 grpc_error* grpc_metadata_batch_add_tail(grpc_metadata_batch* batch,
                                          grpc_linked_mdelem* storage,
                                          grpc_mdelem elem_to_add) {
-  GPR_ASSERT(!GRPC_MDISNULL(elem_to_add));
```

Comment:
I would keep these and make them DEBUG ASSERT.

Reply:
I removed the one for add_tail() since it just calls link_tail() directly which does have the DBG assert. It's fine to add it back, but a tad redundant.